codice/ddf
# DDF-4066 Fixed bug so that templates show under the correct tabs

#### What does this PR do?
Fixes a bug so that templates show only under the 'My Search Forms' for the creator of the template. Also ensures that users that the template has been shared with only has the template show under the 'Shared Templates' tab.

Running the formatter changed 3 files, but the code changes in the PR only concerns itself with forms-sharing/search-form-sharing.collection.js (lines 85-99, so please review that and dismiss the others since it's only changes in formatting).

#### Who is reviewing it? 
@middlets719 
@Schachte 
@gordocanchola 
@adimka 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@jlcsmith
@millerw8
@rzwiefel

#### How should this be tested?
1. Full build and then unzip snapshot.
2. Go into the /etc directory in snapshot and change users.properties to 
```
admin=admin,group,admin,manager,viewer,system-admin,systembundles
localhost=localhost,group,admin,manager,viewer,system-admin,system-history,system-user,systembundles
dummy1=dummy,group,admin,manager,viewer,system-admin,system-history,systembundles
dummy2=dummy,group,manager,viewer,system-history,systembundles
```
and change users.attirbutes to
```
{
    "admin" : {
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress":"admin@localhost.local"
    },
    "localhost" : {
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress":"system@localhost.local"
    },
    "dummy1" : {
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress":"dummy@gmail.com"
    },
    "dummy2" : {
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress":"dummy2@gmail.com"
    }
}
```
3. Open up two different browsers (ie. Chrome and Chrome Incognito) and go to https://localhost:8993/login. 
Login in as `dummy1:dummy` in one browser and
login in as `dummy2:dummy` in another browser.
Make sure to enable Experimental Features in the admin console under the configurations tab.

4. Using dummy1's browser, create another search form.
<img width="943" alt="screen shot 2018-08-16 at 11 07 59 am" src="https://user-images.githubusercontent.com/23177404/44223987-7525e300-a146-11e8-97e7-674b0f03f1d8.png">

Name the form and save it. Repeat this process two more times.

5. Then pick one of the forms and share it. 
Share one of the forms by email. 
Share a different form by Role.
Share the last form by neither Role nor by email.

6. Under dummy1's search forms, 'My Search Forms' should have all the created forms. However under 'Shared Templates' it should not have any of the created forms.

7. Under dummy2's search forms, 'My Search Forms' should not have any of the created forms. They should be seen under the 'Shared Templates'. The only form that should not be seen in the 'Shared Templates' tab is the form you shared without a role and without an email from dummy1.

8. Try different variants of the above instructions and verify sharing works as it should.

#### Any background context you want to provide?
#### What are the relevant tickets?
https://codice.atlassian.net/projects/DDF/issues/DDF-4066?filter=allissues

#### Screenshots
#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
